crates/shim-sgx/src/handler/mod.rs

```diff
@@ -77,17 +77,22 @@ pub struct Handler<'a> {
 }
 
 impl<'a> Handler<'a> {
-    /// Acknowledge pages committed by the host with ENCLS[EAUG].
-    fn accept_mmap(&mut self, region: Region) {
-        assert!(region.start < region.end);
-        let length = region.end - region.start;
-        let secinfo = Class::Regular.info(Flags::READ | Flags::WRITE | Flags::EXECUTE);
+    /// Acknowledge ENCLS[EAUG].
+    fn accept_pending(&mut self, addr: usize, length: c_size_t, prot: c_int) {
+        let flags = Flags::from_bits_truncate((prot | PROT_READ) as u8);
         let zero = PageAddr::from_start_address(VirtAddr::new(ZERO.as_ptr() as u64)).unwrap();
-        for i in 0..length.items() {
-            let virt_addr = VirtAddr::new((region.start.raw() + i * Page::SIZE) as u64);
-            let page_addr = PageAddr::from_start_address(virt_addr).unwrap();
-            // FIXME: handle with `.unwrap()` once `SecInfo` implements `Debug` trait.
-            if secinfo.accept_copy(page_addr, zero).is_err() {
+        for i in 0..length / Page::SIZE {
+            let addr = PageAddr::from_start_address(VirtAddr::new((addr + i * Page::SIZE) as u64))
+                .unwrap();
+            if flags == Flags::READ | Flags::WRITE {
+                if Class::Regular
+                    .info(flags | Flags::PENDING)
+                    .accept(addr)
+                    .is_err()
+                {
+                    panic!();
+                }
```

Comment:
one small patch for the `sgx` crate and you can `.unwrap()` instead of this `.is_err()` and `panic!()`:

```diff
diff --git a/src/page/sinfo.rs b/src/page/sinfo.rs
index 98116b1..a3d6195 100644
--- a/src/page/sinfo.rs
+++ b/src/page/sinfo.rs
@@ -51,7 +51,7 @@ impl From<Class> for SecInfo {
 }
 
 /// Error codes for `SecInfo::accept()`
-#[derive(Copy, Clone, PartialEq, Eq)]
+#[derive(Copy, Clone, PartialEq, Eq, Debug)]
 #[non_exhaustive]
 pub enum AcceptError {
     /// CPU cores have not exited from the previous grace period.
```

Reply:
I have a PR for `sgx` crate containing such commit: https://github.com/enarx/sgx/pull/87. However, I think the deal is that this should land first before making changes to that crate.

Reply:
now using unwrap()